docker/test/integration/minifi/core/DockerTestCluster.py

```diff
@@ -318,3 +318,31 @@ def wait_for_metric_class_on_prometheus(self, metric_class, timeout_seconds):
 
     def wait_for_processor_metric_on_prometheus(self, metric_class, timeout_seconds, processor_name):
         return PrometheusChecker().wait_for_processor_metric_on_prometheus(metric_class, timeout_seconds, processor_name)
+
+    def wait_for_peak_memory_usage(self, minimum_peak_memory_usage: int, timeout_seconds: int) -> bool:
+        start_time = time.perf_counter()
+        current_peak_memory_usage = get_peak_memory_usage(get_minifi_pid())
+        if current_peak_memory_usage is None:
+            logging.warning("Failed to determine peak memory usage")
+            return False
+        while (time.perf_counter() - start_time) < timeout_seconds:
+            if current_peak_memory_usage > minimum_peak_memory_usage:
+                return True
+            current_peak_memory_usage = get_peak_memory_usage(get_minifi_pid())
+            time.sleep(1)
```

Comment:
why do we check `current_peak_memory_usage` twice?  would this work?
```suggestion
        while (time.perf_counter() - start_time) < timeout_seconds:
            current_peak_memory_usage = get_peak_memory_usage(get_minifi_pid())
            if current_peak_memory_usage is None:
                logging.warning("Failed to determine peak memory usage")
                return False
            if current_peak_memory_usage > minimum_peak_memory_usage:
                return True
            time.sleep(1)
```
(similarly in the other function below)

Reply:
good idea, I've changed it in https://github.com/apache/nifi-minifi-cpp/pull/1456/commits/d08d3c78709c3a1042429de7ac18591909f6e4d2#diff-4487753f519234543da32b7926ab6cfc45c79ae2467eca71c8f6c269c7195eefR325-R328

Reply:
good idea, I've changed it in https://github.com/apache/nifi-minifi-cpp/pull/1456/commits/d08d3c78709c3a1042429de7ac18591909f6e4d2#diff-4487753f519234543da32b7926ab6cfc45c79ae2467eca71c8f6c269c7195eefR325-R328

Reply:
good idea, I've changed it in https://github.com/apache/nifi-minifi-cpp/pull/1456/commits/d08d3c78709c3a1042429de7ac18591909f6e4d2#diff-4487753f519234543da32b7926ab6cfc45c79ae2467eca71c8f6c269c7195eefR325-R328